lib/3scale_toolbox/entities/backend_method.rb

```diff
@@ -0,0 +1,88 @@
+module ThreeScaleToolbox
+  module Entities
+    class BackendMethod
+      VALID_PARAMS = %w[friendly_name system_name description].freeze
```

Comment:
By looking at System's API, a required value is the "unit". We should add it into the VALID_PARAMS array

Reply:
This is documentation error from 3scale API. 

3scale create method endpoint does not respect `unit` field and always set to "1". 

3scale API docs do not allow make request to create method without 'unit' field. However using any other http client (ruby client or even curl), you can create methods without 'unit' field.

@3scale/system guys should confirm this

Reply:
it should respect the `unit` field

https://github.com/3scale/porta/blob/master/app/controllers/admin/api/backend_apis/metrics_controller.rb#L120-L122

https://github.com/3scale/porta/blob/master/app/controllers/admin/api/backend_apis/metrics_controller.rb#L63

if not could you please create a JIRA issue with the curl request in the steps to reproduce?

Thank you

Reply:
We can check and fix for ER2 @eguzki

Reply:
@hallelujah same behavior reported for service/product methods. 
https://issues.redhat.com/browse/THREESCALE-2923

Basically, all methods inherit parent "hits" metric unit. If you change `hits` metric unit, methods unit shown in UI will change automatically accordingly.  So it can not be changed. 

I tried, anyway, to see 3scale API behavior. Currently `hits` metric unit is `hit`.

* Create backend method with `unit: blah`, response object does not have unit field
```
<- "POST /admin/api/backend_apis/52466/metrics/2555418433285/methods.json 
<- "{\"friendly_name\":\"my_method_01\",\"unit\":\"blah\",\"description\":\"some description\"}"
-> "HTTP/1.1 201 Created\r\n"
...

-> "{\"method\":{\"id\":2555418433398,\"system_name\":\"my_method_01.52466\",\"friendly_name\":\"my_method_01\",\"description\":\"some description\",\"created_at\":\"2020-01-31T16:45:30+01:00\",\"updated_at\":\"2020-01-31T16:45:30+01:00\",\"links\":[{\"rel\":\"parent\",\"href\":\"https://3scale-supertest-admin.3scale.net/admin/api/backend_apis/52466/metrics/2555418433285\"},{\"rel\":\"self\",\"href\":\"https://3scale-supertest-admin.3scale.net/admin/api/backend_apis/52466/metrics/2555418433285/methods?id=2555418433398\"}]}}"
read 483 bytes
Conn close
---
id: 2555418433398
system_name: my_method_01.52466
friendly_name: my_method_01
description: some description
created_at: '2020-01-31T16:45:30+01:00'
updated_at: '2020-01-31T16:45:30+01:00'
....
```
* Read object using GET endpoint, but `unit` field still is not there.
```
<- "GET /admin/api/backend_apis/52466/metrics/2555418433285/methods/2555418433398.json
.....
-> "{\"method\":{\"id\":2555418433398,\"system_name\":\"my_method_01.52466\",\"friendly_name\":\"my_method_01\",\"description\":\"some description\",\"created_at\":\"2020-01-31T16:45:30+01:00\",\"updated_at\":\"2020-01-31T16:45:30+01:00\",\"links\":[{\"rel\":\"parent\",\"href\":\"https://3scale-supertest-admin.3scale.net/admin/api/backend_apis/52466/metrics/2555418433285\"},{\"rel\":\"self\",\"href\":\"https://3scale-supertest-admin.3scale.net/admin/api/backend_apis/52466/metrics/2555418433285/methods?id=2555418433398\"}]}}"
read 483 bytes
Conn close
---
id: 2555418433398
system_name: my_method_01.52466
friendly_name: my_method_01
description: some description
....
```
* Create new backend method with `unit: other`
```
<- "POST /admin/api/backend_apis/52466/metrics/2555418433285/methods.json 
<- "{\"friendly_name\":\"my_method_02\",\"unit\":\"other\",\"description\":\"some description\"}"
.....
-> "{\"method\":{\"id\":2555418433434,\"system_name\":\"my_method_02.52466\",\"friendly_name\":\"my_method_02\",\"description\":\"some description\",\"created_at\":\"2020-01-31T16:47:54+01:00\",\"updated_at\":\"2020-01-31T16:47:54+01:00\",\"links\":[{\"rel\":\"parent\",\"href\":\"https://3scale-supertest-admin.3scale.net/admin/api/backend_apis/52466/metrics/2555418433285\"},{\"rel\":\"self\",\"href\":\"https://3scale-supertest-admin.3scale.net/admin/api/backend_apis/52466/metrics/2555418433285/methods?id=2555418433434\"}]}}"
read 483 bytes
Conn close
---
id: 2555418433434
system_name: my_method_02.52466
friendly_name: my_method_02
description: some description
...
```

UI showing both methods created

![Screenshot_2020-01-31 Metrics - Index Red Hat 3scale API Management](https://user-images.githubusercontent.com/881529/73553914-0f690c00-444b-11ea-98d9-f06d0480234a.png)

Note that unit shown is `hit`, the `hits` metric unit.

* Update backend method's unit attr to `unit: blah`
```
<- "PUT /admin/api/backend_apis/52466/metrics/2555418433285/methods/2555418433398.json 
<- "{\"unit\":\"blah\"}"
....
-> "{\"method\":{\"id\":2555418433398,\"system_name\":\"my_method_01.52466\",\"friendly_name\":\"my_method_01\",\"description\":\"some description\",\"created_at\":\"2020-01-31T16:45:30+01:00\",\"updated_at\":\"2020-01-31T16:45:30+01:00\",\"links\":[{\"rel\":\"parent\",\"href\":\"https://3scale-supertest-admin.3scale.net/admin/api/backend_apis/52466/metrics/2555418433285\"},{\"rel\":\"self\",\"href\":\"https://3scale-supertest-admin.3scale.net/admin/api/backend_apis/52466/metrics/2555418433285/methods?id=2555418433398\"}]}}"
read 483 bytes
Conn close
---
id: 2555418433398
system_name: my_method_01.52466
friendly_name: my_method_01
description: some description
...
```
Update request did not fail, but nothing changed. 

From the toolbox point of view, no need to set `unit` when creating backend methods, so no issue here